scan/scan.go

```diff
@@ -237,10 +247,43 @@ func (s *sourcescanner) walkSource(conn redis.Conn, path string, f os.FileInfo,
 		return nil, nil
 	}
 
+	var dfData DummyFile
+	dummyFile := GetConfig().DummyFiles
+
 	d := new(filedata)
 	d.path = path[len(GetConfig().Repository):]
-	d.size = f.Size()
-	d.modTime = f.ModTime()
+
+	if dummyFile {
+		raw, err := ioutil.ReadFile(path)
```

Comment:
In case someone makes a configuration error, we don't want mirrorbits to load a multi-gigabyte file into memory. Making a custom reader that will do some basic sanity check would be much better. Even better, if the file is not a "dummyFile", then fallback to the standard method. This way, we can have an hybrid repository.